### PR TITLE
imhex: init at 1.6.0-git

### PR DESCRIPTION
**Pull request by leo60228**
###### Motivation for this change
ImHex has been receiving a lot of attention the past few days, I know the author, and I wanted to try it.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/imhex/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    rev = "e3b5a55eba51aaab1473abf4c2a40975ddd1e9fa";
+    sha256 = "1nfq2yy21chldkg79r9nnb8n7g9fma6c9fnfy7yqgsl6y7iffxqw";
+  };
+
+  hardeningDisable = [ "format" ];
+
+  nativeBuildInputs = [ cmake pkgconfig ];
+  buildInputs = [ nlohmann_json capstone file glfw3 glm jsoncpp llvm_9 openssl python38 ];
+
+  patches = [ ./default-db.patch ];
```
Comment (symphorien) [inline pkgs/applications/editors/imhex/default.nix:19]: please submit the patch upstream, and fetch the corresponding commit (in your fork) with fetchpatch.

**Comment by leo60228** [comment]
This patch shouldn't be upstreamed. I've talked to the developer about this
and not using the default magic database was intended to make it easier to
package on Windows.

**Comment by symphorien** [comment]
then it should be commented copiously.

File: pkgs/applications/editors/imhex/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    owner = "WerWolv";
+    repo = "ImHex";
+    #rev = "v${version}";
+    rev = "e3b5a55eba51aaab1473abf4c2a40975ddd1e9fa";
+    sha256 = "1nfq2yy21chldkg79r9nnb8n7g9fma6c9fnfy7yqgsl6y7iffxqw";
+  };
+
+  hardeningDisable = [ "format" ];
+
+  nativeBuildInputs = [ cmake pkgconfig ];
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/imhex/default.nix:16]:
```suggestion
  nativeBuildInputs = [ cmake pkg-config ];
```

File: pkgs/applications/editors/imhex/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
+{ stdenv, gcc10Stdenv, cmake, pkgconfig, nlohmann_json, capstone, file, glfw3, glm, jsoncpp, llvm_9, openssl, python38, fetchFromGitHub }:
+gcc10Stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/imhex/default.nix:2]:
```suggestion
{ stdenv, cmake, pkg-config, nlohmann_json, capstone, file, glfw3, glm, jsoncpp, llvm_9, openssl, python3, fetchFromGitHub }:

stdenv.mkDerivation rec {
```

File: pkgs/applications/editors/imhex/default.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      repo = "ImHex-Patterns";
+      rev = "f5e5345aa6986ddaeb87a9648f5c539ae38a61a1";
+      sha256 = "1dh77jc8dzc7xvzrabjrspn6lf3w8ab4lddvgpwaq6hd7lkdfpf6";
+    };
+  in ''
+  substituteInPlace source/views/view_pattern.cpp --replace 'patterns' "${patterns}/patterns"
+  substituteInPlace source/lang/preprocessor.cpp --replace 'include/' "${patterns}/includes/"
+  '';
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/imhex/default.nix:35]:
```suggestion
  meta = with lib; {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Also please change the version to 1.6.0

**Comment by bobvanderlinden** [comment]
Another attempt has been made at https://github.com/NixOS/nixpkgs/pull/184488

**Pull request closed without merging**